packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -3493,6 +3497,62 @@ class RenderSemanticsAnnotations extends RenderProxyBox {
       markNeedsSemanticsUpdate();
   }
 
+  /// The handler for [SemanticsAction.accessibilityFocus].
+  ///
+  /// This handler is invoked when the accessibility focus is moved
+  /// to the node annotated with this handler. The accessibility focus is the
+  /// green (on Android with  TalkBack) or black (on iOS with VoiceOver)
+  /// rectangular show on screen to indicate with what element an accessibility
+  /// user is currently interacting with.
+  ///
+  /// The accessibility focus is different from the input focus. The input focus
+  /// is usually held by the element that currently response to keyboard inputs.
+  /// Accessibility focus and input focus can be held by two different nodes!
+  ///
+  /// See also:
+  ///
+  ///  * [onLoseAccessibilityFocus], which is invoked when the accessibility
+  ///    focus is removed from the node
+  ///  * [FocusNode], [FocusScope], [FocusManager], which manage the input focus
+  VoidCallback get onAccessibilityFocus => _onAccessibilityFocus;
+  VoidCallback _onAccessibilityFocus;
+  set onAccessibilityFocus(VoidCallback handler) {
+    if (_onAccessibilityFocus == handler)
+      return;
+    final bool hadValue = _onAccessibilityFocus != null;
+    _onAccessibilityFocus = handler;
+    if ((handler != null) != hadValue)
+      markNeedsSemanticsUpdate();
+  }
+
+  /// The handler for [SemanticsAction.loseAccessibilityFocus].
+  ///
+  /// This handler is invoked when the accessibility focus is moved
+  /// away from the node annotated with this handler. The accessibility focus is
+  /// the green (on Android with  TalkBack) or black (on iOS with VoiceOver)
+  /// rectangular show on screen to indicate with what element an accessibility
+  /// user is currently interacting with.
+  ///
+  /// The accessibility focus is different from the input focus. The input focus
+  /// is usually held by the element that currently response to keyboard inputs.
+  /// Accessibility focus and input focus can be held by two different nodes!
+  ///
+  /// See also:
+  ///
+  ///  * [onAccessibilityFocus], which is invoked when the node gains
```

Comment:
nit: prefer using consistent terminology. In other parts you say "focus moves to/away from node" while here it's "node gains". FWIW, I prefer "gain"/"lose" terminology (which is also why I suggest renaming to "didGain" and "didLose").